### PR TITLE
Make the code compute the diffuse fraction by default

### PR DESCRIPTION
**Pull request by Whyborn**
# CABLE

## Description

The ```CABLE-POP_TRENDY``` branch of the code currently sets ```cable_user%calc_fdiff=.FALSE.``` by default. The only place this was modified was in the CRU meteorology setup, so would effectively break anyone else's simulation not using this workflow who was unaware of this change.

This change changes the default to ```cable_user%calc_fdiff=.TRUE.```, so that the code computes the direct and diffuse fractions via the spitter function. The check used in the CRU/TRENDY has been moved to initialisation. If the user turns off diffuse fraction calculation (via the ```ReadDiffFrac=.TRUE.``` option in the CRU namelist), but doesn't also provide a dataset to read from, the code will now throw an error.

This is not an ideal fix, as we should be checking against the ```cable_user%calc_fdiff``` option, rather than the ```CRU%ReadDiffFrac``` option. I think a better option is to remove the ```CRU%ReadDiffFrac``` option altogether and only have the ```cable_user%calc_fdiff```, given they effectively contain identical information. That would also require minor adjustments to the TRENDY scripts.

Fixes at least part of the problems discussed in #526.

## Type of change

- [x] Bug fix

### CODE REVIEW

**Review by har917 — APPROVED** [review]
@Whyborn Code changes look okay - we'll need some testing (short TRENDY runs before/after at the minimum) to demonstrate that this doesn't break something inadvertently (but I can't see how).

**Comment by juergenknauer** [comment]
It looks good to me as well. I would also support to only use one of the flags. I am not sure which one though. One could argue that this is a met input thing and it would be better placed in the cru namelist.

File: offline/cable_cru_TRENDY.F90 [hunk 1]
```diff
@@ -163,6 +163,18 @@ SUBROUTINE CRU_INIT(CRU)
...
+    cable_user%calc_fdiff = .FALSE.
+    ! Make sure we have a dataset for fdiff
+    IF (TRIM(InputFiles(fdiff)) == "None") THEN
+      WRITE(ERROR_UNIT,*) "Specified to read diffuse fraction from"//&
+        " file, but no diffuse fraction file given."
+      ERROR STOP 1
+    END IF
+  ELSE
+    cable_user%calc_fdiff = .TRUE.
+  END IF
```
Comment (ccarouge) [inline offline/cable_cru_TRENDY.F90:177]: Wouldn't this be clearer:

```suggestion
  ! Calculate diffusive fraction if we are not reading it in.
  cable_user%calc_fdiff = .NOT. CRU%ReadDiffFrac
  
  ! Make sure we have a dataset for fdiff if set to read in
  IF (CRU%ReadDiffFrac .AND. (TRIM(InputFiles(fdiff)) == "None")) THEN
    WRITE(ERROR_UNIT,*) "Specified to read diffuse fraction from"//&
      " file, but no diffuse fraction file given."
    ERROR STOP 1
  END IF
```

**Comment by ccarouge** [comment]
> It looks good to me as well. I would also support to only use one of the flags. I am not sure which one though. One could argue that this is a met input thing and it would be better placed in the cru namelist.

It is a met related thing but not limited to the CRU met forcings, so we can't just have it in the CRU namelist.

**Comment by Whyborn** [comment]
I have made the change that drops back to using strictly the ```cable_user%calc_fdiff```. All instances of ```CRU%ReadDiffFrac``` have been removed. The way to set this variable should be through ```cable.nml```, so if this route is chosen, there will need to be some minor updates to the TRENDY configuration.

The only changes would be to ```run_cable.sh```:

1. Swap the order of the logical assignment [here](https://github.com/CABLE-LSM/TRENDY/blob/28c299cc9adef088dc9db2bf3eb0818ea039bdde/run_cable.sh#L274-L279)
2. Remove ```ReadDiffFrac``` from ```cru.nml``` [here](https://github.com/CABLE-LSM/TRENDY/blob/28c299cc9adef088dc9db2bf3eb0818ea039bdde/run_cable.sh#L307)
3. Add ```cable_user%calc_fdiff=${fdiff_bool}``` to the ```cable.nml``` template [here](https://github.com/CABLE-LSM/TRENDY/blob/28c299cc9adef088dc9db2bf3eb0818ea039bdde/run_cable.sh#L333-L375)

Let me know your thoughts- I can go back to the previous commit (+ suggestions from Claire) if that would be preferred.

**Comment by ccarouge** [comment]
I'm happy with the proposed solution, but I'm not using the TRENDY configuration. @juergenknauer @mcuntz do you have any objections to the proposed changes in the namelists?

**Comment by juergenknauer** [comment]
It all looks good to me. Just note that I had to branch off from the main TRENDY branch because the current version does not work with my setup. After this year's TRENDY runs we can try and merge them again and hopefully I can use the new setup next year.

**Comment by Whyborn** [comment]
Branch off the TRENDY configuration, or branch off the TRENDY branch of the CABLE code?

**Comment by juergenknauer** [comment]
branch off the code, because the problem lies somewhere in there. Unfortunately I couldn't see where yet. I would need to checkout after each commit, compile, and see if it works or not.
